File: ST02ContatoAutoComparaTest.java
import org.junit.Test;
import org.junit.Before;
import org.junit.After;
import static org.junit.Assert.*;
import static org.hamcrest.CoreMatchers.is;
import static org.hamcrest.core.IsNot.not;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.remote.RemoteWebDriver;
import org.openqa.selenium.remote.DesiredCapabilities;
import org.openqa.selenium.Dimension;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.interactions.Actions;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.openqa.selenium.JavascriptExecutor;
import org.openqa.selenium.Alert;
import org.openqa.selenium.Keys;
import java.util.*;
import java.net.MalformedURLException;
import java.net.URL;
public class ST02ContatoAutoComparaTest {
  private WebDriver driver;
  private Map<String, Object> vars;
  JavascriptExecutor js;
  @Before
  public void setUp() {
    driver = new ChromeDriver();
    js = (JavascriptExecutor) driver;
    vars = new HashMap<String, Object>();
  }
  @After
  public void tearDown() {
    driver.quit();
  }
  @Test
  public void sT02ContatoAutoCompara() {
    driver.get("https://www.autocompara.com.br/");
    driver.manage().window().setSize(new Dimension(1058, 678));
    driver.findElement(By.linkText("Fale conosco")).click();
    js.executeScript("window.scrollTo(0,50)");
    js.executeScript("window.scrollTo(0,213)");
    driver.findElement(By.cssSelector(".ng-tns-c59-0:nth-child(4)")).click();
    driver.findElement(By.cssSelector("#mat-option-0 > .mat-option-text")).click();
    driver.findElement(By.id("mat-input-1")).click();
    driver.findElement(By.id("mat-input-2")).click();
    driver.findElement(By.id("mat-input-3")).click();
    driver.findElement(By.id("mat-input-4")).click();
    driver.findElement(By.cssSelector(".ng-tns-c59-7 > .mat-form-field-infix")).click();
    driver.findElement(By.cssSelector(".cdk-overlay-backdrop")).click();
    driver.findElement(By.cssSelector(".ng-tns-c64-10 > .mat-select-arrow")).click();
    driver.findElement(By.cssSelector(".cdk-overlay-backdrop")).click();
    driver.findElement(By.cssSelector(".ng-tns-c64-12:nth-child(2)")).click();
    driver.findElement(By.cssSelector(".cdk-overlay-backdrop")).click();
    driver.findElement(By.cssSelector(".ng-tns-c64-12:nth-child(2)")).click();
    driver.findElement(By.cssSelector(".cdk-overlay-backdrop")).click();
    driver.findElement(By.cssSelector(".ng-tns-c64-12 > .mat-select-arrow")).click();
    driver.findElement(By.cssSelector(".cdk-overlay-backdrop")).click();
    driver.findElement(By.cssSelector(".ng-tns-c59-6 > .mat-form-field-infix")).click();
    driver.findElement(By.cssSelector("ac-textarea")).click();
    {
      List<WebElement> elements = driver.findElements(By.cssSelector(".ng-tns-c59-2 > .mat-error"));
      assert(elements.size() > 0);
    }
    {
      List<WebElement> elements = driver.findElements(By.cssSelector(".ng-tns-c59-3 > .mat-error"));
      assert(elements.size() > 0);
    }
    {
      List<WebElement> elements = driver.findElements(By.cssSelector(".ng-tns-c59-4 > .mat-error"));
      assert(elements.size() > 0);
    }
    assertThat(driver.findElement(By.cssSelector(".ng-tns-c59-5 > .mat-error")).getText(), is("Campo obrigatório."));
    {
      List<WebElement> elements = driver.findElements(By.cssSelector(".ng-tns-c59-5 > .mat-error"));
      assert(elements.size() > 0);
    }
    assertThat(driver.findElement(By.cssSelector(".ng-tns-c59-7 > .mat-form-field-infix")).getText(), is("  Seguradora *"));
    {
      List<WebElement> elements = driver.findElements(By.id("mat-error-6"));
      assert(elements.size() > 0);
    }
    assertThat(driver.findElement(By.cssSelector(".ng-tns-c64-10 > .mat-select-placeholder")).getText(), is(" "));
    {
      List<WebElement> elements = driver.findElements(By.id("mat-error-7"));
      assert(elements.size() > 0);
    }
    assertThat(driver.findElement(By.cssSelector(".ng-tns-c59-11 > .mat-form-field-infix")).getText(), is("  Tipo de documento"));
    assertThat(driver.findElement(By.id("mat-input-0")).getText(), is("  "));
    {
      List<WebElement> elements = driver.findElements(By.cssSelector(".ng-tns-c59-6 > .mat-error"));
      assert(elements.size() > 0);
    }
    driver.findElement(By.id("mat-input-1")).click();
    driver.findElement(By.id("mat-input-1")).sendKeys("Leandro Lima Da Rocha");
    driver.findElement(By.id("mat-input-2")).click();
    driver.findElement(By.id("mat-input-2")).sendKeys("316.186.518-97");
    driver.findElement(By.cssSelector(".ng-tns-c59-4:nth-child(4)")).click();
    driver.findElement(By.id("mat-input-3")).sendKeys("(11) 9 3471-5462");
    driver.findElement(By.id("mat-input-4")).click();
    driver.findElement(By.id("mat-input-4")).sendKeys("lrocha1983@gmail.com");
    driver.findElement(By.cssSelector(".ng-tns-c64-8 > .mat-select-placeholder")).click();
    driver.findElement(By.cssSelector("#mat-option-11 > .mat-option-text")).click();
    driver.findElement(By.cssSelector(".ng-tns-c64-10:nth-child(2)")).click();
    driver.findElement(By.cssSelector("#mat-option-14 > .mat-option-text")).click();
    driver.findElement(By.id("mat-input-5")).click();
    driver.findElement(By.id("mat-input-5")).sendKeys("frv-6930");
    driver.findElement(By.cssSelector(".mat-select-placeholder")).click();
    driver.findElement(By.cssSelector("#mat-option-16 > .mat-option-text")).click();
    driver.findElement(By.cssSelector(".ng-tns-c59-14:nth-child(4)")).click();
    driver.findElement(By.id("mat-input-6")).sendKeys("123456789088555555");
    driver.findElement(By.id("mat-input-0")).click();
    driver.findElement(By.id("mat-input-0")).sendKeys("Boa tarde, gostaria de saber da minha apólice...");
    driver.findElement(By.cssSelector(".ng-star-inserted > button")).click();
    {
      List<WebElement> elements = driver.findElements(By.cssSelector(".ng-star-inserted:nth-child(2) > div > .ng-star-inserted"));
      assert(elements.size() > 0);
    }
    {
      List<WebElement> elements = driver.findElements(By.cssSelector("div:nth-child(4) button"));
      assert(elements.size() > 0);
    }
    driver.findElement(By.cssSelector("div:nth-child(4) button")).click();
    js.executeScript("window.scrollTo(0,0)");
  }
}

File: ST02ContatoAutoComparaTest.cs
using System;
using System.Collections;
using System.Collections.Generic;
using System.Linq;
using System.Threading;
using OpenQA.Selenium;
using OpenQA.Selenium.Chrome;
using OpenQA.Selenium.Firefox;
using OpenQA.Selenium.Remote;
using OpenQA.Selenium.Support.UI;
using OpenQA.Selenium.Interactions;
using NUnit.Framework;
[TestFixture]
public class ST02ContatoAutoComparaTest {
  private IWebDriver driver;
  public IDictionary<string, object> vars {get; private set;}
  private IJavaScriptExecutor js;
  [SetUp]
  public void SetUp() {
    driver = new ChromeDriver();
    js = (IJavaScriptExecutor)driver;
    vars = new Dictionary<string, object>();
  }
  [TearDown]
  protected void TearDown() {
    driver.Quit();
  }
  [Test]
  public void sT02ContatoAutoCompara() {
    driver.Navigate().GoToUrl("https://www.autocompara.com.br/");
    driver.Manage().Window.Size = new System.Drawing.Size(1058, 678);
    driver.FindElement(By.LinkText("Fale conosco")).Click();
    js.ExecuteScript("window.scrollTo(0,50)");
    js.ExecuteScript("window.scrollTo(0,213)");
    driver.FindElement(By.CssSelector(".ng-tns-c59-0:nth-child(4)")).Click();
    driver.FindElement(By.CssSelector("#mat-option-0 > .mat-option-text")).Click();
    driver.FindElement(By.Id("mat-input-1")).Click();
    driver.FindElement(By.Id("mat-input-2")).Click();
    driver.FindElement(By.Id("mat-input-3")).Click();
    driver.FindElement(By.Id("mat-input-4")).Click();
    driver.FindElement(By.CssSelector(".ng-tns-c59-7 > .mat-form-field-infix")).Click();
    driver.FindElement(By.CssSelector(".cdk-overlay-backdrop")).Click();
    driver.FindElement(By.CssSelector(".ng-tns-c64-10 > .mat-select-arrow")).Click();
    driver.FindElement(By.CssSelector(".cdk-overlay-backdrop")).Click();
    driver.FindElement(By.CssSelector(".ng-tns-c64-12:nth-child(2)")).Click();
    driver.FindElement(By.CssSelector(".cdk-overlay-backdrop")).Click();
    driver.FindElement(By.CssSelector(".ng-tns-c64-12:nth-child(2)")).Click();
    driver.FindElement(By.CssSelector(".cdk-overlay-backdrop")).Click();
    driver.FindElement(By.CssSelector(".ng-tns-c64-12 > .mat-select-arrow")).Click();
    driver.FindElement(By.CssSelector(".cdk-overlay-backdrop")).Click();
    driver.FindElement(By.CssSelector(".ng-tns-c59-6 > .mat-form-field-infix")).Click();
    driver.FindElement(By.CssSelector("ac-textarea")).Click();
    var elements = driver.FindElements(By.CssSelector(".ng-tns-c59-2 > .mat-error"));
    Assert.True(elements.Count > 0);
    var elements = driver.FindElements(By.CssSelector(".ng-tns-c59-3 > .mat-error"));
    Assert.True(elements.Count > 0);
    var elements = driver.FindElements(By.CssSelector(".ng-tns-c59-4 > .mat-error"));
    Assert.True(elements.Count > 0);
    Assert.That(driver.FindElement(By.CssSelector(".ng-tns-c59-5 > .mat-error")).Text, Is.EqualTo("Campo obrigatório."));
    var elements = driver.FindElements(By.CssSelector(".ng-tns-c59-5 > .mat-error"));
    Assert.True(elements.Count > 0);
    Assert.That(driver.FindElement(By.CssSelector(".ng-tns-c59-7 > .mat-form-field-infix")).Text, Is.EqualTo("  Seguradora *"));
    var elements = driver.FindElements(By.Id("mat-error-6"));
    Assert.True(elements.Count > 0);
    Assert.That(driver.FindElement(By.CssSelector(".ng-tns-c64-10 > .mat-select-placeholder")).Text, Is.EqualTo(" "));
    var elements = driver.FindElements(By.Id("mat-error-7"));
    Assert.True(elements.Count > 0);
    Assert.That(driver.FindElement(By.CssSelector(".ng-tns-c59-11 > .mat-form-field-infix")).Text, Is.EqualTo("  Tipo de documento"));
    Assert.That(driver.FindElement(By.Id("mat-input-0")).Text, Is.EqualTo("  "));
    var elements = driver.FindElements(By.CssSelector(".ng-tns-c59-6 > .mat-error"));
    Assert.True(elements.Count > 0);
    driver.FindElement(By.Id("mat-input-1")).Click();
    driver.FindElement(By.Id("mat-input-1")).SendKeys("Leandro Lima Da Rocha");
    driver.FindElement(By.Id("mat-input-2")).Click();
    driver.FindElement(By.Id("mat-input-2")).SendKeys("316.186.518-97");
    driver.FindElement(By.CssSelector(".ng-tns-c59-4:nth-child(4)")).Click();
    driver.FindElement(By.Id("mat-input-3")).SendKeys("(11) 9 3471-5462");
    driver.FindElement(By.Id("mat-input-4")).Click();
    driver.FindElement(By.Id("mat-input-4")).SendKeys("lrocha1983@gmail.com");
    driver.FindElement(By.CssSelector(".ng-tns-c64-8 > .mat-select-placeholder")).Click();
    driver.FindElement(By.CssSelector("#mat-option-11 > .mat-option-text")).Click();
    driver.FindElement(By.CssSelector(".ng-tns-c64-10:nth-child(2)")).Click();
    driver.FindElement(By.CssSelector("#mat-option-14 > .mat-option-text")).Click();
    driver.FindElement(By.Id("mat-input-5")).Click();
    driver.FindElement(By.Id("mat-input-5")).SendKeys("frv-6930");
    driver.FindElement(By.CssSelector(".mat-select-placeholder")).Click();
    driver.FindElement(By.CssSelector("#mat-option-16 > .mat-option-text")).Click();
    driver.FindElement(By.CssSelector(".ng-tns-c59-14:nth-child(4)")).Click();
    driver.FindElement(By.Id("mat-input-6")).SendKeys("123456789088555555");
    driver.FindElement(By.Id("mat-input-0")).Click();
    driver.FindElement(By.Id("mat-input-0")).SendKeys("Boa tarde, gostaria de saber da minha apólice...");
    driver.FindElement(By.CssSelector(".ng-star-inserted > button")).Click();
    var elements = driver.FindElements(By.CssSelector(".ng-star-inserted:nth-child(2) > div > .ng-star-inserted"));
    Assert.True(elements.Count > 0);
    var elements = driver.FindElements(By.CssSelector("div:nth-child(4) button"));
    Assert.True(elements.Count > 0);
    driver.FindElement(By.CssSelector("div:nth-child(4) button")).Click();
    js.ExecuteScript("window.scrollTo(0,0)");
  }
}
